homeassistant/components/amberelectric/config_flow.py

```diff
@@ -0,0 +1,141 @@
+        self._api_token = None
+
+        if user_input is not None:
+            token = user_input[CONF_API_TOKEN]
+            self._sites = await self.hass.async_add_executor_job(
+                self._fetch_sites, token
+            )
+
+            if self._sites is not None:
+                self._api_token = token
+                return await self.async_step_site()
+
+        else:
+            user_input = {CONF_API_TOKEN: ""}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_API_TOKEN, default=user_input[CONF_API_TOKEN]
+                    ): str,
+                }
+            ),
+            errors=self._errors,
+        )
+
+    async def async_step_site(self, user_input: dict[str, Any] = None):
+        """Step to select site."""
+        self._errors = {}
+
+        api_token = self._api_token
+        if user_input is not None:
+            if self._sites is None:
+                return self.async_show_form(
+                    step_id="user",
+                    data_schema=vol.Schema(
+                        {
+                            vol.Required(
+                                CONF_API_TOKEN, default=user_input[CONF_API_TOKEN]
+                            ): str,
+                        }
+                    ),
+                    errors={CONF_API_TOKEN: "no_site"},
+                )
+
+            site_nmi = user_input[CONF_SITE_NMI]
+            sites = list(filter(lambda site: site.nmi == site_nmi, self._sites))
+
+            if len(sites) != 0:
+                site: Site = sites[0]
+                site_id = site.id
+                name = user_input.get(CONF_SITE_NAME, site_id)
+                return self.async_create_entry(
+                    title=name,
+                    data={
+                        CONF_SITE_ID: site_id,
+                        CONF_API_TOKEN: api_token,
+                        CONF_SITE_NMI: site.nmi,
+                    },
+                )
+        else:
+            user_input = {
+                CONF_API_TOKEN: api_token,
+                CONF_SITE_NMI: "",
+                CONF_SITE_NAME: "",
+            }
+
+            if self._sites is None:
+                return self.async_show_form(
+                    step_id="user",
+                    data_schema=vol.Schema(
+                        {
+                            vol.Required(CONF_API_TOKEN, default=api_token): str,
+                        }
+                    ),
+                    errors={CONF_API_TOKEN: "no_site"},
+                )
+
+        return self.async_show_form(
+            step_id="site",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SITE_NMI, default=user_input[CONF_SITE_NMI]
+                    ): vol.In(list(map(lambda site: site.nmi, self._sites))),
```

Comment:
`map` code is hard to read.
Prefer this:
```suggestion
                    ): vol.In([site.nmi for site in self._sites]),
```